### ISSUE TITLE
On a self-hosted site with WooCommerce and Jetpack connection, Products tab does not work as expected when Secure Sign On is enabled

### ISSUE THREAD

**Issue by pehaa** [Automattic/wp-calypso#72429]
### Quick summary

On a self-hosted site with WooCommerce and Jetpack connection, Products tab in the menu area redirects to the **Products** page in WP-Admin (`/wp-admin/edit.php?post_type=product`). 
However, when Secure Sign On is enabled, this is no longer the case. The `Products` tab points to `https://wordpress.com/types/product/{site}` page that displays the _You need permission to manage this post type
Ask your site administrator to grant you access_ notice.

### Steps to reproduce

1. Set up a self-hosted site with Jetpack connection and WooCommerce installed
2. Visit the site in Calypso
3. Try to access **Products** from the main menu (left sidebar)
4. You will be redirected to your site's WP Admin > Products (`/wp-admin/edit.php?post_type=product`)
<img width="405" alt="Markup 2023-01-23 at 13 38 34" src="https://user-images.githubusercontent.com/774920/214057242-0d7b0f13-76f6-44e4-bdb2-f942bad96b59.png">

5. Now, activate Secure Sign On (In your WP dashboard in **Jetpack > Settings > Security > WordPress.com Login** or in Calypso **Settings > Security > WordPress.com log in**
6. Reload the browser in Calypso
7. You will notice that the `gridicons-external` icon disappeared
8. When you click the **Products** you will be pointed to `https://wordpress.com/types/product/{site}` page that displays _You need permission to manage this post type Ask your site administrator to grant you access_
<img width="1094" alt="Markup 2023-01-23 at 13 40 55" src="https://user-images.githubusercontent.com/774920/214057320-6a76ce73-ad9b-4b3f-85b8-2b3e58965c63.png">

### What you expected to happen

The icon does not disappear and the user is redirected to their WP Admin > Products.
In other words, I would expect that the experience does not change whether SSO is enabled or not.
The notice is also confusing, it's displayed regardless if there are only one or multiple admins on the site.

### What actually happened

User is confused. They are admin but don't have permission to edit/see products.

### Context

Reported in 5863777-zd-woothemes

### Platform (Simple, Atomic, or both?)

Self-hosted

### Reproducibility

Consistent

### Severity

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### Workaround details

- Clarify to the user that they should always log in to their WP admin in order to edit Products. 
- Disable SSO to remove the confusion

**Comment by darcie** [comment]
Another report in 5990734-zen

**Comment by cuemarie** [comment]
📌 SCRUBBING : RESULT - *Replicated / Could Not Replicate / Uncertain*
- Tested on Atomic – Could Not Replicate
- Tested on SelfHosted Jetpack-Connected Site - **Replicated**
- Replicable outside of DotCom - Yes 

📌 FINDINGS/SCREENSHOTS/VIDEO
- Replicated as described. It seems a quick solution would be to have the `Products` item link to it's wp-admin page `/wp-admin/edit.php?post_type=product` like the items in that section of Calypso (Feedback, Orders, Coupons, & WP Admin)

**Screenshots/Recordings**
![Markup on 2023-04-11 at 16:35:37](https://user-images.githubusercontent.com/27249804/231310955-a81f6bf1-7df3-4d19-93c0-33894a5045b3.png)

📌 ACTIONS
- Assigned to Non-Urgent

**Comment by JenSwish** [comment]
Another report in 6340575-zen

**Comment by bobmatyas** [comment]
Reported on 7179189-zen

**Comment by davemart-in** [comment]
This may only affect a small number of customers, but for those it does affect it's a blocker. I reprioritized this as "High".

**Comment by valterlorran** [comment]
I started proposing a fix for this here: https://github.com/Automattic/jetpack/pull/40840
